lib/testing_utils.py

```
@@ -0,0 +1,87 @@
+import time
+from typing import Callable, TypeVar, ParamSpec
```

Ruff (0.15.7)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)